src/snowflake/snowpark/session.py

```diff
@@ -602,6 +599,26 @@ def __init__(
                 DEFAULT_COMPLEXITY_SCORE_UPPER_BOUND,
             ),
         )
+
+        self._thread_safe_session_enabled: bool = (
+            self._conn._get_client_side_session_parameter(
+                _PYTHON_SNOWPARK_ENABLE_THREAD_SAFE_SESSION, False
+            )
+        )
+        self._thread_store = (
+            threading.local() if self._thread_safe_session_enabled else dict()
```

Comment:
instead of dict, maybe just use optional here, and only use the _thread_store value if the parameter is on

Reply:
I would then have to create something like `__analyzer` and `__conn` since the `_analyzer` is already created as a property now. I think since we have tests passing, it should be okay. I am using a `DummyThreadLocal` class now. Everything will be removed once I rollout parameter.